src/index.ts

```diff
@@ -90,6 +91,7 @@ const recommended = {
     'sf-plugin/no-default-and-depends-on-flags': 'error',
     'sf-plugin/only-extend-SfCommand': 'warn',
     'sf-plugin/spread-base-flags': 'warn',
+    'sf-plugin/no-depends-on-boolean-flag': 'warn',
```

Comment:
warn instead of error because there are cases where depending on a boolean flag works as expected:
* `flag1` is boolean and has no default value/no `allowNo`
* `flag2` depends on `flag1`

there's no way `flag1` can have a value if it's not passed in. Still, the warning should push users to double-check and future-proof their code 😉